server/lomas_server/utils/collection_models.py

```diff
@@ -88,11 +89,156 @@ class DatasetsCollection(BaseModel):
     ]
 
 
+class ColumnMetadata(BaseModel):
+    """Base model for column metadata"""
+
+    private_id: bool = False
+    nullable: bool = False
+    # TODO create validator and test these, see issue #323
+    max_partition_length: Optional[Annotated[int, Field(gt=0)]] = None
+    max_influenced_partitions: Optional[Annotated[int, Field(gt=0)]] = None
+    max_partition_contributions: Optional[Annotated[int, Field(gt=0)]] = None
+
+
+class StrMetadata(ColumnMetadata):
+    """Model for string metadata"""
+
+    type: Literal["string"]
+
+
+class CategoricalColumnMetadata(ColumnMetadata):
+    """Model for categorical column metadata"""
+
+    @model_validator(mode="after")
+    def validate_categories(self):
+        """Makes sure number of categories matches cardinality."""
+        if len(self.categories) != self.cardinality:
+            raise ValueError(
+                "Number of categories should be equal to cardinality."
+            )
+        return self
+
+
+class StrCategoricalMetadata(CategoricalColumnMetadata):
+    """Model for categorical string metadata"""
+
+    type: Literal["string"]
+    cardinality: int
+    categories: List[str]
+
+
+class Precision(IntEnum):
+    """Precision of integer and float data"""
+
+    SINGLE = 32
+    DOUBLE = 64
+
+
+class BoundedColumnMetadata(ColumnMetadata):
+    """Model for columns with bounded data"""
+
+    @model_validator(mode="after")
+    def validate_bounds(self):
+        """Validates column bounds."""
+        if (
+            self.lower is not None
+            and self.upper is not None
+            and self.lower > self.upper
+        ):
+            raise ValueError("Lower bound cannot be larger than upper bound.")
```

Comment:
Actually not InternalServerError but maybe Collection error or somthing like this

Reply:
same here